common/configuration/asm_follower.go

```diff
@@ -0,0 +1,97 @@
+package configuration
+
+import (
+	"context"
+	"fmt"
+	"net/url"
+	"strings"
+
+	"connectrpc.com/connect"
+	ftlv1 "github.com/TBD54566975/ftl/backend/protos/xyz/block/ftl/v1"
+	"github.com/TBD54566975/ftl/backend/protos/xyz/block/ftl/v1/ftlv1connect"
+	"github.com/alecthomas/types/optional"
+)
+
+// asmFollower uses AdminService to get/set secrets from the leader
+type asmFollower struct {
+	client ftlv1connect.AdminServiceClient
+}
+
+var _ asmClient = &asmFollower{}
+
+func (f *asmFollower) list(ctx context.Context) ([]Entry, error) {
+	module := ""
+	includeValues := false
+	resp, err := f.client.SecretsList(ctx, connect.NewRequest(&ftlv1.ListSecretsRequest{
+		Module:        &module,
+		IncludeValues: &includeValues,
+	}))
+	if err != nil {
+		return nil, err
+	}
+	entries := []Entry{}
+	for _, s := range resp.Msg.Secrets {
+		components := strings.Split(s.RefPath, ".")
```

Comment:
I've seen this pattern a few times, maybe a `fn (rp *) ToRef (Ref, error) {}` ?

Reply:
I'll make a helper func for this 👍

Reply:
`ParseRef` already exists 🤦‍♂️